packages/svelte/src/compiler/migrate/index.js

```diff
@@ -457,10 +466,80 @@ const instance_script = {
 				state.str.prependLeft(start, '$state(');
 				state.str.appendRight(end, ')');
 			} else {
-				state.str.prependLeft(
-					/** @type {number} */ (declarator.id.typeAnnotation?.end ?? declarator.id.end),
-					' = $state()'
+				/**
+				 * @type {AssignmentExpression | undefined}
+				 */
+				let assignment_in_labeled;
+				/**
+				 * @type {LabeledStatement | undefined}
+				 */
+				let labeled_statement;
+
+				const possible_derived = bindings.every((binding) =>
+					binding.references.every((reference) => {
+						const declaration_idx = reference.path.findIndex(
+							(el) => el.type === 'VariableDeclaration'
+						);
+						const assignment_idx = reference.path.findIndex(
+							(el) => el.type === 'AssignmentExpression'
+						);
+						const update_idx = reference.path.findIndex((el) => el.type === 'UpdateExpression');
+						const labeled_idx = reference.path.findIndex(
+							(el) => el.type === 'LabeledStatement' && el.label.name === '$'
+						);
+
+						if (assignment_idx !== -1 && labeled_idx !== -1) {
+							if (assignment_in_labeled) return false;
+							assignment_in_labeled = /** @type {AssignmentExpression} */ (
+								reference.path[assignment_idx]
+							);
+							labeled_statement = /** @type {LabeledStatement} */ (reference.path[labeled_idx]);
+						}
+
+						return (
+							update_idx === -1 &&
+							(declaration_idx !== -1 ||
+								(labeled_idx !== -1 && assignment_idx !== -1) ||
+								(labeled_idx === -1 && assignment_idx === -1))
```

Comment:
This might be a bit simpler if you're not searching for the index but rather the object directly. Not as much `-1`, and no `reference.path[idx]` above

Reply:
Yeah i usually go with indices just because this just stores a reference to a number instead of an object but that's silly in this case.